consensus/src/processes/pruning_proof/build.rs

```diff
@@ -335,33 +420,34 @@ impl PruningProofManager {
         &self,
         root: Hash,
         selected_tip: Hash,
-        ghostdag_store: &Arc<DbGhostdagStore>,
+        transient_ghostdag_store: &Arc<DbGhostdagStore>,
         required_block: Option<Hash>,
         level: BlockLevel,
+        transient_relations_store: &Arc<RwLock<DbRelationsStore>>,
         ghostdag_k: KType,
     ) -> bool {
-        let relations_service = RelationsStoreInFutureOfRoot {
-            relations_store: self.level_relations_services[level as usize].clone(),
+        let transient_relations_service = RelationsStoreInFutureOfRoot {
+            relations_store: transient_relations_store.read().clone(),
```

Comment:
this line should be 
```
relations_store: MTRelationsService::new(transient_relations_store.clone())
```

cloning the inner store is not the way to go (effectively you are deep-cloning it in a way that it can be used without the lock)